### DISCUSSION TITLE
PageLogFormat with"job-actual-printer-uri" or current IPP variable name

### DISCUSSION THREAD

**Discussion by rwiesler** [OpenPrinting/cups#1002] [Q&A] [answered]
Current: CUPS 2.2.10 / Debian 10
Planned: CUPS 2.4.02 / Debian 12
 
Hi,
 
there exists a printer class with three memembers for load-balancing and from time to time one of the printers fails and the printjob in general has to be restarted.

It is tidious to find out wich printjob on the printer failed exactly and the default string for the `PageLogFormat` in `cupsd.conf` does not contain a variable for it.
So I did a little research and found in [Mailing list ARChives / Re: [cups.general] Question on use of Class from 2007-08-16 20:00:34](https://marc.info/?l=cups&m=118729447530719&w=2) the hint for the variable `%{job-actual-printer-uri}`.
 
```
# cupsd.conf adapted line
PageLogFormat %p %u %j %T %P %C %{job-billing} %{job-originating-host-name} %{job-name} %{media} %{sides} %{job-actual-printer-uri}
```

But neither `%{job-actual-printer-uri}` or `%{job-printer-uri}` or `%{member-names}` or `%{member-uris}` lists the actual used printer of the failed printjob.
 
I might got things wrong but which IPP varible would I use to get a log entry for the failed job on the printer?

Thanks for your help.

**Comment by michaelrsweet** [comment, marked as answer]
OK, so looking at the current code you want "{job-printer-uri-actual}" to capture which printer in a class is printing the job.
